app/policies/test_result_policy.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+# Policies for accessing TestResults
+class TestResultPolicy < ApplicationPolicy
+  def index?
+    Pundit.policy(user, record.profile) || Pundit.policy(user, record.host)
+  end
+
+  def show?
+    Pundit.policy(user, record.profile) || Pundit.policy(user, record.host)
+  end
+
+  # Only show TestResults belonging to profiles owned by the current user
+  class Scope < ::ApplicationPolicy::Scope
+    def resolve
+      available_profiles = ProfilePolicy::Scope.new(user, ::Profile)
+      available_hosts = HostPolicy::Scope.new(user, ::Host)
+      scope.where(profile_id: available_profiles.resolve).or(
```

Comment:
This works exclusively through scopes, instead of using `pluck`, `resolve` returns the AR relation without resolving it (yes, the naming is a bit strange) - the final SQL called is:
```
SELECT "test_results".* FROM "test_results" WHERE ("test_results"."profile_id" IN (SELECT "profiles"."id" FROM "profiles" WHERE "profiles"."account_id" = 'b428b5d9-df19-5bb9-a1dc-115e071b836c') OR "test_results"."host_id" IN (SELECT "hosts"."id" FROM "hosts" WHERE "hosts"."account_id" = 'b428b5d9-df19-5bb9-a1dc-115e071b836c'))
```